src/app/screens/book-list/book-list.component.html

```diff
@@ -1,9 +1,12 @@
-<div class="list-container">
-  <ng-template ngFor let-book [ngForOf]="books" let-i="index">
-    <div class="list-item">
-      <a class="item-book-cover"></a>
-      <span class="item-bold-text">{{book.title}}</span>
-      <span class="item-text">{{book.author}}</span>
-    </div>
-  </ng-template>
+<div class="container">
+  <input [(ngModel)]="filter" #ctrl="ngModel" class="filter" type="text" name="filter" >
+  <div class="list-container">
+    <ng-template ngFor let-book [ngForOf]="books | titleFilter:filter" let-i="index">
```

Comment:
When did you learned this weird way of loop? change please and you dont need index, also delete ng-template

Reply:
In official documentation https://angular.io/api/common/NgForOf

Reply:
😆